### ISSUE TITLE
issue with Tide

### ISSUE THREAD

**Issue by jcanderan** [compomics/searchgui#291]
Does Tide still work with SearchGUI? (linux 4.0.22)?

I keep getting the following error message: 

`FATAL: Error reading index (/dirpath/SearchGUI-4.0.22/resources/temp/search_engines/tide/fasta-index/pepix)`

As a note, this file doesn't exist. I see a file called protix in the directory.

An example full log below:

```
2021-03-22 00:01:28 /dirpath/SearchGUI-4.0.22/resources/Tide/linux/linux_64bit/crux tide-index /dirpath/jc_referencetide_concatenated_target_decoy.fasta /dirpath/SearchGUI-4.0.22/resources/temp/search_engines/tide/fasta-index --temp-dir /dirpath/SearchGUI-4.0.22/resources/temp/search_engines/tide --overwrite T --mods-spec C+57.02146372057,2M+15.99491461956 --max-mods 255 --decoy-format none --output-dir /dirpath/SearchGUI-4.0.22/resources/temp/search_engines/tide --peptide-list F --verbosity 30 --min-length 6 --max-length 30 --min-mass 200.0 --max-mass 7200.0 --isotopic-mass mono --clip-nterm-methionine F --enzyme trypsin --missed-cleavages 2 --digestion partial-digest
2021-03-22 00:01:28
2021-03-22 00:01:28 Mon Mar 22 00:01:28 UTC 2021 Indexing jc_referencetide_concatenated_target_decoy.fasta for Tide.
2021-03-22 00:01:28
2021-03-22 00:01:28 INFO: Beginning tide-index.
2021-03-22 00:01:28 WARNING: The output directory '/dirpath/SearchGUI-4.0.22/resources/temp/search_engines/tide' already exists.
2021-03-22 00:01:28 Existing files will be overwritten.
2021-03-22 00:01:28 INFO: CPU: bhaal
2021-03-22 00:01:28 INFO: Crux version: 3.0.17109
2021-03-22 00:01:28 INFO: Mon 22 Mar 2021 12:01:28 AM UTC
2021-03-22 00:01:28 INFO: Running tide-index...
2021-03-22 00:01:28 INFO: Writing results to output directory '/dirpath/SearchGUI-4.0.22/resources/temp/search_engines/tide/fasta-index'.
2021-03-22 00:01:28 INFO: Reading /dirpath/referencetide/fasta/jc_referencetide_concatenated_target_decoy.fasta and computing unmodified peptides...
2021-03-22 01:33:28
2021-03-22 01:33:28
2021-03-22 01:33:28 Mon Mar 22 01:33:28 UTC 2021 Tide Indexing finished for jc_referencetide_concatenated_target_decoy.fasta (1 hour 32 minutes 512.0 milliseconds).
2021-03-22 01:33:28
2021-03-22 01:33:28
2021-03-22 01:33:28 Mon Mar 22 01:33:28 UTC 2021 Importing spectrum files.
2021-03-22 01:33:29 Mon Mar 22 01:33:29 UTC 2021 Importing spectrum files completed (305.0 milliseconds).
2021-03-22 01:33:29
2021-03-22 01:33:29
2021-03-22 01:33:29 Processing: filename.mgf (1/1)
2021-03-22 01:33:29
2021-03-22 01:33:29 Mon Mar 22 01:33:29 UTC 2021 Converting spectrum file filename.mgf for Tide.
2021-03-22 01:33:37 10% 20% 30% 40% 50% 60% 70% 80% 90%
2021-03-22 01:33:37
2021-03-22 01:33:37 tide search command:
2021-03-22 01:33:37 /dirpath/SearchGUI-4.0.22/resources/Tide/linux/linux_64bit/crux tide-search --compute-sp F /dirpath/SearchGUI-4.0.22/resources/temp/peak_lists/filename.ms2 /dirpath/SearchGUI-4.0.22/resources/temp/search_engines/tide/fasta-index --overwrite T --top-match 10 --precursor-window 10.0 --precursor-window-type ppm --fileroot filename --exact-p-value F --output-dir /dirpath/SearchGUI-4.0.22/resources/temp/search_engines/tide/crux-output --spectrum-min-mz 0.0 --min-peaks 20 --spectrum-charge all --max-precursor-charge 5 --remove-precursor-peak F --remove-precursor-tolerance 1.5 --print-search-progress 1000 --use-flanking-peaks F --use-neutral-loss-peaks F --mz-bin-width 0.02 --mz-bin-offset 0.0 --num-threads 20 --concat F --txt-output T --sqt-output F --pepxml-output F --mzid-output F --pin-output F --verbosity 30
2021-03-22 01:33:37
2021-03-22 01:33:37
2021-03-22 01:33:37 Mon Mar 22 01:33:37 UTC 2021 Processing filename.ms2 with Tide.
2021-03-22 01:33:37
2021-03-22 01:33:37 FATAL: Error reading index (/dirpath/SearchGUI-4.0.22/resources/temp/search_engines/tide/fasta-index/pepix)
2021-03-22 01:33:37
2021-03-22 01:33:37
2021-03-22 01:33:37 Mon Mar 22 01:33:37 UTC 2021 Tide Searching finished for filename.ms2 (29.0 milliseconds).
2021-03-22 01:33:37
2021-03-22 01:33:37 Mon Mar 22 01:33:37 UTC 2021 Error: Source '/dirpath/SearchGUI-4.0.22/resources/temp/search_engines/tide/crux-output/filename.tide-search.target.txt' does not exist
2021-03-22 01:33:37 Mon Mar 22 01:33:37 UTC 2021 An error occurred while running SearchGUI. Please contact the developers.
2021-03-22 01:33:37 Mon Mar 22 01:33:37 UTC 2021 The search or processing did not finish properly!
```

Corresponding message in SearchGUI.log:

```
Mon Mar 22 00:01:27 UTC 2021: SearchGUI version 4.0.22.
Memory given to the Java virtual machine: 51539607552.
Total amount of memory in the Java virtual machine: 51539607552.
Free memory: 51514439680.
Java version: 14.0.2.
java.io.FileNotFoundException: Source '/dirpath/SearchGUI-4.0.22/resources/temp/search_engines/tide/crux-output/filename.tide-search.target.txt' does not exist
        at org.apache.commons.io.FileUtils.moveFile(FileUtils.java:2995)
        at eu.isas.searchgui.SearchHandler$SearchWorker.doInBackground(SearchHandler.java:3044)
        at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
        at java.base/java.lang.Thread.run(Thread.java:832)
```

Thanks!

**Comment by jcanderan** [comment]
Similarly, getting the following error message with Andromeda (linux SearchGUI 4.0.22) a few seconds the command is executed:

```
2021-03-22 06:20:29 Mon Mar 22 06:20:29 UTC 2021 Andromeda configuration.
2021-03-22 06:20:29
2021-03-22 06:20:45 Mon Mar 22 06:20:45 UTC 2021 Error: conf/databases.xml (No such file or directory)
2021-03-22 06:20:45 Mon Mar 22 06:20:45 UTC 2021 An error occurred while running SearchGUI. Please contact the developers.
2021-03-22 06:20:45 Mon Mar 22 06:20:45 UTC 2021 The search or processing did not finish properly!
```

**Comment by hbarsnes** [comment]
> Does Tide still work with SearchGUI? (linux 4.0.22)?

As far as I know it should still work on Linux. And at least it still works fine on Windows.

Are you running SearchGUI from the command line? If so, please make sure that you are using complete paths when referring to files or folders. 

> Similarly, getting the following error message with Andromeda (linux SearchGUI 4.0.22) a few seconds the command is executed

I'm afraid Andromeda version in SearchGUI is only supported on Windows.

**Comment by jcanderan** [comment]
Yes -- command line and I am referring to complete paths. I had edited the log to just put dirpath instead of the full path. The file "pepix" does not exist in the folder if I list the files. Only the file "protix" does.

Anyway, I tried using a CRAP FASTA (from GPM) to test and it works now so I assume it's a tide issue and not a SearchGUI issue. Essentially, with a larger uniprot FASTA, tide index writes protix but not pepix, but with the smaller FASTA, it writes both.

I don't need tide for any specific purpose so I'm not going to investigate further, but I was going through the available engines to see what works and what the results look like as I had some unexpected results with using MSGF + Xtandem together (but individually they work fine and also work fine adding OMSSA as a third engine).

Thanks!
